sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2097,7 +2157,7 @@ case class UnBase64(child: Expression)
   """,
   since = "1.5.0")
 // scalastyle:on line.size.limit
-case class Decode(bin: Expression, charset: Expression)
+case class StringDecode(bin: Expression, charset: Expression)
```

Comment:
Do we have an alternative for the previous `decode` after this change?

Reply:
This only renames the expression, not the function. This PR just adds an overload of the `decode` function when there are more than 3 parameters.

Reply:
I'd suggest the following followup:
1. add a new function name for `StringDecode`. Probably "string_decode".
2. require `decode` function to take more than 2 parameters, under ansi mode(or a legacy config)